In [1]:
import numpy as np
from scipy import stats
from matplotlib import pyplot as plt
plt.style.use('seaborn-whitegrid')
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

### Задача 1
Даны значения величины заработной платы заемщиков банка (salary) и значения их поведенческого кредитного скоринга (scoring):

salary = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]

scoring = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

1. Найдите ковариацию этих двух величин: сначала без использования специальных функций, а затем с помощью функции numpy.cov. Полученные значения должны быть равны.
2. Найдите коэффициент корреляции Пирсона: с помощью ковариации и средних квадратических отклонений двух признаков, а затем с помощью специальных функций из библиотек numpy и pandas.

#### Решение:


In [2]:
df = pd.DataFrame({'salary':[35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
      'scoring':[401, 574, 874, 919, 459, 739, 653, 902, 746, 832]
     })

In [3]:
X = df['salary']
Y = df['scoring']

MX = X.mean()
MY = Y.mean()

cov = ((X - MX) * (Y - MY)).sum() / (X.shape[0] - 1)
cov

10175.37777777778

In [4]:
np.cov(X, Y)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [5]:
df[['salary', 'scoring']].corr()

salary  scoring
salary   1.00000  0.88749
scoring  0.88749  1.00000

In [6]:
corr = cov / (X.std() * Y.std(ddof = 1))
corr

0.8874900920739164

In [7]:
np.corrcoef(X, Y)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

#### Ответ:
1. Ковариация равна 10175.38
2. Коэффициент корреляции Пирсона равен 0.88749

### Задача 2
Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111

Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

#### Решение:


In [8]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
n = iq.shape[0]
mean = iq.mean()
std = iq.std(ddof=1)
p = 0.95
alpha = 1 - p

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)
(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))

(110.55608365158724, 125.64391634841274)

#### Ответ:


### Задача 3
Известно, что рост футболистов в сборной распределен нормально с известной дисперсией 25. На выборке объёма 27 выборочное среднее составило 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

#### Решение:


In [9]:
p = 0.95
alpha = 1 - p
n = 27
mean = 174.2
std = 25
alpha = 1 - p
t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)
(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))

(164.77011888269857, 183.6298811173014)

#### Ответ:
Доверительный интервал роста футболистов от 164,77 до 183,63 см.

### Задача 4
Выберите тему для проектной работы по курсу Теории вероятностей и математической статистики и напишите ее в комментарии к Практическому заданию.

#### Решение:


#### Ответ:
